components/admin_console/admin_console.jsx

```diff
@@ -483,6 +484,11 @@ export default class AdminConsole extends React.Component {
                                             schema: AdminDefinition.settings.customization.legal_and_support.schema,
                                         }}
                                     />
+                                    <SCRoute
+                                        path={`${props.match.url}/custom_terms_of_service`}
+                                        component={CustomTermsOfServiceSettings}
```

Comment:
Why does this require custom stuff. Why can't we just use the declarative system console?

Reply:
We need the terms of service text to be fetched and saved separately from the admin config. This was needed due to some complexity in the server side of the code.

https://github.com/mattermost/mattermost-server/pull/9450#pullrequestreview-158208485

For phase 1, we were doing this in redux itself and abstracting this complexity from the web app.
Now we have changed this as part of removing this redux "hack".

https://mattermost.atlassian.net/browse/MM-12414

Reply:
@crspeller If it would help, here's the link to the related discussion on mattermost: https://community.mattermost.com/core/pl/aj9k9ekkjtg7zmnasmswmcfaxr

Reply:
Yes. Well this is just probably unnecessary complexity that should have been contained in a plugin. Not really much we can do about it now.